python/dlr/tf_model.py

```diff
@@ -0,0 +1,143 @@
+
+        self.model_path = model_path
+        self.logger = create_logger(log_file=error_log_file)
+        if dev_type is not None:
+            devices = ["cpu", "gpu"]
+            if dev_type not in devices:
+                raise ValueError(
+                    "Invalid device type {}. Valid devices: {}".format(dev_type, devices))
+        if dev_type == 'gpu':
+            physical_devices = tf.config.list_physical_devices('GPU')
+            tf.config.experimental.set_memory_growth(
+                physical_devices[dev_id], True)
+            self.logger.info("TF memory_growth: {}".format(
+                tf.config.experimental.get_memory_growth(physical_devices[dev_id])))
+
+        self.use_default_dlr = use_default_dlr
+        self._lib = None
+        self.version = VERSION
+
+        self.saved_model = tf.saved_model.load(model_path)
+        tag_set = 'serve'
+        assert len(self.saved_model.signatures) > 0
+        signature_def_key = 'serving_default'
+        if signature_def_key not in self.saved_model.signatures:
+            signature_def_key = list(self.saved_model.signatures.keys())[0]
+        self.func = self.saved_model.signatures[signature_def_key]
+        meta_graph_def = saved_model_utils.get_meta_graph_def(model_path, tag_set)
+        inputs_tensor_info = meta_graph_def.signature_def[signature_def_key].inputs
+        outputs_tensor_info = meta_graph_def.signature_def[signature_def_key].outputs
+        self.input_tensor_names = list(inputs_tensor_info.keys())
+        self.output_tensor_names = list(outputs_tensor_info.keys())
+
+    def get_input_names(self):
+        """
+        Get all input names
+        Returns
+        -------
+        out : list of :py:class:`str`
+        """
+        return self.input_tensor_names
+
+    def get_output_names(self):
+        """
+        Get all output names
+        Returns
+        -------
+        out : list of :py:class:`str`
+        """
+        return self.output_tensor_names
+
+    def get_input(self, name, shape=None):
+        """
+        Get the current value of an input
+        Parameters
+        ----------
+        name : str
+            The name of an input
+        shape : np.array (optional)
+            If given, use as the shape of the returned array. Otherwise, the shape of
+            the returned array will be inferred from the last call to set_input().
+        """
+        self._validate_input_name(name)
+        if name not in self.input_values:
+            return None
+        out = self.input_values[name]
+        if shape is not None:
+            out = out.reshape(shape)
+        return out
+
+    def _validate_input_name(self, name):
+        if name not in self.input_tensor_names:
+            raise ValueError(
+                "Invalid input tensor name '{}'. List of input tensor names: {}".format(name, self.input_tensor_names))
```

Comment:
`Invalid input name '{}'. List of input names: {}`

Reply:
done